src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpCodeActions.cs

```diff
@@ -281,6 +281,146 @@ public int Y2
             Assert.Equal(expectedText, VisualStudio.Editor.GetText());
         }
 
+        [CriticalWpfTheory]
+        [InlineData(FixAllScope.Project)]
+        [InlineData(FixAllScope.Solution)]
+        [Trait(Traits.Feature, Traits.Features.CodeActionsFixAllOccurrences)]
+        [WorkItem(33507, "https://github.com/dotnet/roslyn/issues/33507")]
+        public void FixAllOccurrencesIgnoresGeneratedCode(FixAllScope scope)
+        {
+            var markup = @"
+using System;
+using $$System.Threading;
+
+class C
+{
+    public IntPtr X1 { get; set; }
+}";
+            var expectedText = @"
+using System;
+
+class C
+{
+    public IntPtr X1 { get; set; }
+}";
+            var generatedSourceMarkup = @"// <auto-generated/>
+using System;
+using $$System.Threading;
+
+class D
+{
+    public IntPtr X1 { get; set; }
+}";
+            var expectedGeneratedSource = @"// <auto-generated/>
+using System;
+
+class D
+{
+    public IntPtr X1 { get; set; }
+}";
+
+            MarkupTestFile.GetPosition(generatedSourceMarkup, out var generatedSource, out int generatedSourcePosition);
+
+            VisualStudio.SolutionExplorer.AddFile(new ProjectUtils.Project(ProjectName), "D.cs", generatedSource, open: false);
+
+            // Switch to the main document we'll be editing
+            VisualStudio.SolutionExplorer.OpenFile(new ProjectUtils.Project(ProjectName), "Class1.cs");
+
+            // Verify that applying a Fix All operation does not change generated files.
+            // This is a regression test for correctness with respect to the design.
+            SetUpEditor(markup);
+            VisualStudio.WaitForApplicationIdle(CancellationToken.None);
+            VisualStudio.Editor.InvokeCodeActionList();
+            VisualStudio.Editor.Verify.CodeAction(
+                "Remove Unnecessary Usings",
+                applyFix: true,
+                fixAllScope: scope);
+
+            Assert.Equal(expectedText, VisualStudio.Editor.GetText());
+
+            VisualStudio.SolutionExplorer.OpenFile(new ProjectUtils.Project(ProjectName), "D.cs");
+            Assert.Equal(generatedSource, VisualStudio.Editor.GetText());
+
+            // Verify that a Fix All in Document in the generated file still does nothing.
+            // ⚠ This is a statement of the current behavior, and not a claim regarding correctness of the design.
```

Comment:
Is this worthy of filing a tracking design issue or do we want to wait on customer feedback on this known broken behavior before spending any cycles on it?

Reply:
And how hard is this to just fix? :smile:

Reply:
> Is this worthy of filing a tracking design issue or do we want to wait on customer feedback on this known broken behavior before spending any cycles on it?

#2753